utils/python/CIME/case.py

```diff
@@ -159,6 +159,9 @@ def copy(self, newcasename, newcaseroot, newcimeroot=None, newsrcroot=None):
         return newcase
 
     def flush(self, flushall=False):
+        if not os.path.isdir(self._caseroot):
```

Comment:
If flush is called (from expect) before the caseroot directory is created the wrong error message is output.